testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseResource.java

```diff
@@ -68,6 +68,28 @@ public void addMessage(final String message) throws IOException {
        eventSink.send(event);
    }
    
+   @POST
+   @Path("/addMessageAndDisconnect")
+   public void addMessageAndDisconnect(final String message) throws IOException, InterruptedException {
+        //clear events store first
+        eventsStore.clear();
```

Comment:
This should go into a syncronized (eventStore) block, or we need to use a thread safe collection for the eventStore.

Reply:
This could not be necessary for the test purpose and here are the reasons: 
1. This is to force empty the evens store. 
2. evensStore is only read after eventSink is closed and eventSource is reconnected with a Last-Event-ID header.

Reply:
OK, makes sense